e2e/cypress/integration/ui/infra-proxy/chef-servers.spec.ts

```diff
@@ -0,0 +1,127 @@
+      cy.get('app-notification.info chef-icon').click();
+
+      cy.contains(serverName).should('exist');
+    });
+
+    it('lists servers', () => {
+      cy.get('[data-cy=add-server-button]').contains('Add Chef Server');
+
+      cy.get('#servers-table-container chef-th').contains('Name');
+      cy.get('#servers-table-container chef-th').contains('FQDN');
+      cy.get('#servers-table-container chef-th').contains('IP Address');
+      cy.get('#servers-table-container chef-th').contains('Number Of Orgs');
+    });
+
+    it('can create a chef server with a custom ID', () => {
+      cy.get('[data-cy=add-server-button]').contains('Add Chef Server').click();
+      cy.get('app-chef-servers-list chef-modal').should('exist');
+      cy.get('[data-cy=add-name]').type(serverName);
+      cy.get('[data-cy=add-id]').should('not.be.visible');
+      cy.get('[data-cy=edit-button]').contains('Edit ID').click();
+      cy.get('[data-cy=id-label]').should('not.be.visible');
+      cy.get('[data-cy=add-id]').should('be.visible').clear().type(customServerID);
+      cy.get('[data-cy=add-fqdn]').type(serverFQDN);
+      cy.get('[data-cy=add-ip-address]').type(serverIP);
+
+      cy.get('[data-cy=add-button]').click();
+      cy.get('app-chef-servers-list chef-modal').should('not.be.visible');
+
+      // verify success notification and then dismiss it
+      // so it doesn't get in the way of subsequent interactions
+      cy.get('app-notification.info').should('be.visible');
+      cy.get('app-notification.info chef-icon').click();
+
+      cy.contains(serverName).should('exist');
+    });
+
+    it('fails to create a chef server with a duplicate ID', () => {
+      cy.get('[data-cy=add-server-button]').contains('Add Chef Server').click();
+      cy.get('app-chef-servers-list chef-modal').should('exist');
+      cy.get('[data-cy=add-name]').type(serverName);
+      cy.get('[data-cy=add-id]').should('not.be.visible');
+      cy.get('[data-cy=edit-button]').contains('Edit ID').click();
+      cy.get('[data-cy=id-label]').should('not.be.visible');
+      cy.get('[data-cy=add-id]').should('be.visible').clear().type(customServerID);
+      cy.get('[data-cy=add-fqdn]').type(serverFQDN);
+      cy.get('[data-cy=add-ip-address]').type(serverIP);
+      cy.get('[data-cy=add-button]').click();
+      cy.get('app-chef-servers-list chef-modal chef-error').contains('already exists')
+        .should('be.visible');
+
+      //  here we exit with the chef-modal exit button in the top right corner
+      cy.get('app-chef-servers-list chef-modal chef-button.close').first().click();
+    });
+
+    it('can cancel creating a chef server', () => {
+      cy.get('[data-cy=add-server-button]').contains('Add Chef Server').click();
+      cy.get('app-chef-servers-list chef-modal').should('exist');
+
+      cy.get('chef-button').contains('Cancel').should('be.visible').click();
+
+      // here we exit with the Cancel button
```

Comment:
Looks like comment is on the wrong line.
```suggestion
      // here we exit with the Cancel button
      cy.get('chef-button').contains('Cancel').should('be.visible').click();
```